### DISCUSSION TITLE
Short Circuiting with Actions

### DISCUSSION THREAD

**Discussion by msmucker** [microsoft/RulesEngine#173] [Q&A] [answered]
I'm trying to use the RulesEngine to run these rules until one succeeds and then execute the Actions for the "winning" rule.  I found a response to a previous question about making the rules short circuit when one succeeds, but my actions don't seem to be executing for child rules.  Is there a configuration what would allow for both short circuiting rules and executing the actions on child rules?

<pre>
{
	"WorkflowName": "TestRuleSet",
	"Rules": [
		{
			"RuleName": "CheckState",
			"Operator": "Or",
			"Enabled": true,
			"ErrorType": "Warning",
			"RuleExpressionType": "LambdaExpression",
			"Rules": [
				{
					"RuleName": "Rule1",
					"ErrorMessage": "Error when evaluating Rule1",
					"Enabled": true,
					"ErrorType": "Warning",
					"RuleExpressionType": "LambdaExpression",
					"Expression": "input1.Vehicle.Make == \"NotFord\"",
					"Actions": {
						"onSuccess": {
							"Name": "OutputExpression",
							"Context": {
								"Expression": "ResultActions.SendEmail(\"8a9d18c6-7541-4939-ab5a-12835ea1bf93\")"
							}
						}
					},
					"SuccessEvent": "ResultActions.SendEmail(\"8a9d18c6-7541-4939-ab5a-12835ea1bf93\")"
				},
				{
					"RuleName": "Rule2",
					"ErrorMessage": "Error when evaluating Rule2",
					"Enabled": true,
					"ErrorType": "Warning",
					"RuleExpressionType": "LambdaExpression",
					"Expression": "input1.Vehicle.Make == \"Ford\"",
					"Actions": {
						"onSuccess": {
							"Name": "OutputExpression",
							"Context": {
								"Expression": "ResultActions.CreateTask(1)"
							}
						}
					},
					"SuccessEvent": "ResultActions.CreateTask(1)"
				}
			]
		}
	]
}
</pre>

**Comment by abbasc52** [comment, marked as answer]
@msmucker running actions for child rules has not been implemented yet. It is being tracked as a part of this issue - https://github.com/microsoft/RulesEngine/issues/89

**Reply by msmucker** [reply]
Thanks for the reply!  Is there an ETA on when that would be completed with issue 89?

**Reply by abbasc52** [reply]
@msmucker this issue has been fixed v3.3.0

**Reply by msmucker** [reply]
Thank you!
